modules/lib/core/index.d.ts

```diff
@@ -0,0 +1,79 @@
+declare global {
+    interface XpXData {
+    }
+}
+
+export type UserKey = `user:${string}:${string}`;
+export type GroupKey = `group:${string}:${string}`;
+export type RoleKey = `role:${string}`;
+
+export type PrincipalKey = UserKey | GroupKey | RoleKey;
+
+export type WorkflowState = 'IN_PROGRESS' | 'PENDING_APPROVAL' | 'REJECTED' | 'READY';
+
+export type WorkflowCheckState = 'PENDING' | 'REJECTED' | 'APPROVED';
+
+export type ContentInheritType = 'CONTENT' | 'PARENT' | 'NAME' | 'SORT';
+
+export interface Workflow {
+    state: WorkflowState;
+    checks?: Record<string, WorkflowCheckState>;
+}
+
+export interface Attachment {
+    name: string;
+    label?: string;
+    size: number;
+    mimeType: string;
+}
+
+export interface PublishInfo {
+    from?: string;
+    to?: string;
+    first?: string;
+}
+
+export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
+    config: Config;
+    descriptor: string;
+    path: string;
+    type: 'page' | 'layout' | 'part';
```

Comment:
Shouldn't this be a union type as well? Similar to `WorkflowCheckState` etc.

Reply:
Not necessary. I would rather think of moving some of other union types from separate declaration to property type.
Component's type is not something that will be frequently imported or used apart from `Component` itself. It can also be retrieved via `type ComponentType = Component['type']` anytime.